src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/arrays/ArraySizeFunctionIterator.java

```diff
@@ -43,6 +43,9 @@ public Item next() {
             this._hasNext = false;
             RuntimeIterator arrayIterator = this._children.get(0);
```

Comment:
This is another example of when you can/should move the array==null test as well as the two lines above to the end of an overriden open() method (after the super call). That way, if array is tested to be the Java null, you can set this._hasNext to false already when opening the iterator and next() won't even be called by the consumer.

In many iterators, a similar thing needs to be done, i.e., move some code from next() to an overriden open() in order to know right away whether an empty sequence is returned or not. This approach is very standard in pull-based querying engines (Zorba does it as well).